ui/src/callWithTelemetryAndErrorHandling.ts

```diff
@@ -228,25 +228,33 @@ function handleTelemetry(context: types.IActionContext, callbackId: string, star
 
             const errorProps: string[] = Object.keys(context.telemetry.properties).filter(key => /(error|exception|stack)/i.test(key));
             // Note: The id of the extension is automatically prepended to the given callbackId (e.g. "vscode-cosmosdb/")
-            ext._internalReporter.sendTelemetryErrorEvent(handlerContext.callbackId, context.telemetry.properties, context.telemetry.measurements, errorProps);
+            ext._internalReporter.sendTelemetryErrorEvent(getTelemetryEventName(handlerContext), context.telemetry.properties, context.telemetry.measurements, errorProps);
         }
     } catch {
-        sendHandlerFailedEvent(context, callbackId, 'telemetry');
+        sendHandlerFailedEvent(handlerContext, 'telemetry');
     }
 }
 
 function shouldSendTelemtry(context: types.IActionContext): boolean {
     return !context.telemetry.suppressAll && !(context.telemetry.suppressIfSuccessful && context.telemetry.properties.result === 'Succeeded');
 }
 
-function sendHandlerFailedEvent(context: types.IActionContext, callbackId: string, handlerName: string) {
+function sendHandlerFailedEvent(context: types.IHandlerContext, handlerName: string) {
     // Errors in our handler logic should not be shown to the user
     try {
         if (shouldSendTelemtry(context)) {
             // Try to send a simple event that should at least alert us that there's a problem
-            ext._internalReporter.sendTelemetryErrorEvent(callbackId, { handlerFailed: handlerName });
+            ext._internalReporter.sendTelemetryErrorEvent(getTelemetryEventName(context), { handlerFailed: handlerName });
         }
     } catch {
         // something must be really wrong with telemetry. just give up
     }
 }
+
+function getTelemetryEventName(context: types.IHandlerContext): string {
+    let suffix: string = '';
+    if (context.telemetry.eventVersion) {
+        suffix = `V${context.telemetry.eventVersion}`;
+    }
+    return context.callbackId + suffix;
+}
```

Comment:
nit: could use ternary here

```suggestion
function getTelemetryEventName(context: types.IHandlerContext): string {
    const suffix = context.telemetry.eventVersion ? `V${context.telmetry.eventVersion}` : '';
    return context.callbackId + suffix;
}
```


or if you're into one-liners

```suggestion
function getTelemetryEventName(context: types.IHandlerContext): string {
    return context.telemetry.eventVersion ? `${context.callbackId}V${context.telmetry.eventVersion}` : context.callbackId;
}```